docs/proposals/apb-state-support.md

```diff
@@ -0,0 +1,60 @@
+# Abstract
+
+This proposal will outline a simple and secure method for allowing APB developers to set state during a APB
+action and then be able to access that state during a subsequent APB action.
+
+## Actions
+   - provision
+   - deprovision
+   - update
+   - bind 
+   - unbind
+   
+   
+## Problem Description
+
+APBs are stateless. All state is managed for them by the broker. Currently the broker passes in parameters specified
+during a request to the catalog. It also passes in some additional parameters such as the namespace etc, additionally 
+we also pass in credentials created during the provision. While this is useful, there is no mechanism for an APB to store 
+and access data across actions without exposing it to the end user. We want to avoid APB developers working around this 
+limitation by doing suboptimal things such as storing extra data in the credentials or by creating a ConfigMap in the users 
+namespace as both of these expose the data to the user and as the namespace is controlled by the user, it is naturally 
+not trustworthy or reliable.
+
+
+
+# Proposed Solution
+
+## APB module for handling state
+
+This module will expose an API to the APB developer for setting and getting key value pairs.
+
+```
+
+- name: Save some stuff
+  asb_set_state:
+    service_name: "{{ service_name }}"
+
+```
+
+Under the hood this APB module would take the key value pair, and store it in a
```

Comment:
Can we decide on secret or configmap? I don't think secrets are meant for large amounts of key value pairs, but we probably want all the application data to be hidden.  IMO let's go with a secret until we have a need to change.

Reply:
I'm not certain to what extent we want this data to be hidden. Given that the configmap would exist in the transient namespace and not the target namespace, it sounds sufficiently hidden. With this being configuration data, I vote for it being a configmap.

Reply:
Yeah my instinct was also a configmap

Reply:
I think I agree with @rthallisey, why not use secrets? Or maybe we have a `store_secret_state` and `store_state` variants?

Reply:
I like the `store_secret_state`, this way`PostgreSQL` would never want to return w/ provision time credentials because they are admin credentials. They could store them in this state to make the subsequent bind calls functional. 

I do think that this creates a larger scope for this PR. If we have two variants of saving state, I could see a first phase and then a second phase IMO.